src/client/config.rs

```diff
@@ -107,9 +116,27 @@ impl Config {
     /// On production setting, the certificate should be added to the local key
     /// storage, using this setting is potentially dangerous.
     ///
-    /// - Defaults to `false`.
+    /// - Defaults to `default`.
     pub fn trust_cert(&mut self) {
-        self.trust_cert = true;
+        if let TrustConfig::CaCertificateLocation(_) = &self.trust {
+            event!(
+                Level::WARN,
+                "trust_ca() was called before. Certificate validation will be disabled.",
```

Comment:
I would instead just panic here. It's the safest option, the warnings might go unnoticed, or be confusing.